service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement;
+import uk.gov.hmcts.reform.fpl.utils.HearingDateHelper;
+
+import java.time.LocalDate;
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Service
+public class DraftCMOService {
+
+    private final ObjectMapper mapper;
+    private final DateFormatterService dateFormatterService;
+
+    @Autowired
+    public DraftCMOService(DateFormatterService dateFormatterService, ObjectMapper mapper) {
+        this.mapper = mapper;
+        this.dateFormatterService = dateFormatterService;
+    }
+
+    public DynamicList getHearingDatesDynamic(CaseDetails caseDetails) {
+        Map<String, Object> caseDataMap = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDataMap, CaseData.class);
+
+        List<Element<HearingBooking>> hearingDetails = caseData.getHearingDetails();
+
+        DynamicList hearingDatesDynamic = buildDynamicListFromHearingDetails(hearingDetails);
+        CaseManagementOrder caseManagementOrder = caseData.getCaseManagementOrder();
+
+        prePopulateHearingDateSelection(hearingDetails, hearingDatesDynamic, caseManagementOrder);
+
+        return hearingDatesDynamic;
+    }
+
+    private void prePopulateHearingDateSelection(
+        List<Element<HearingBooking>> hearingDetails,
+        DynamicList hearingDatesDynamic,
+        CaseManagementOrder caseManagementOrder) {
+        if (caseManagementOrder != null) {
+            String hearingDateId = ObjectUtils.isEmpty(caseManagementOrder) ? ""
+                : caseManagementOrder.getHearingDateId();
+            if (!isEmpty(hearingDateId)) {
+                // There was a previous hearing date therefore we need to remap it
+                String date = hearingDetails.stream()
+                    .filter(Objects::nonNull)
+                    .filter(element -> element.getId().toString().equals(hearingDateId))
+                    .findFirst()
+                    .map(element -> convertDate(element.getValue().getDate()))
+                    .orElse("");
```

Comment:
what happens when the else is hit here, you give empty string as a date in the dynamic list. Does this break anything?

Reply:
No it will not break anything. it will just select the blank as discussed.